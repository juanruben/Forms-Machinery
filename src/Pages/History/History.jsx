import React, { Component } from 'react';
// import { StateContext } from '../State';
import {
    Document, Page, Text, View, StyleSheet, PDFViewer, PDFDownloadLink,
} from '@react-pdf/renderer';
import {
    Modal, ModalHeader, ModalBody, ModalFooter,
} from 'reactstrap';
import ReactTable from 'react-table';
import TopBar from '../../Components/TopBar/TopBar';
import IconButton from '../../Components/IconButton/IconButton';
import { tableConfig, dummyData } from '../../config';

class History extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showing: false,
            ready: false,
        };

        this.toggle = this.toggle.bind(this);
    }

    toggle() {
        this.setState((prevState) => ({
            showing: !prevState.showing,
            ready: false,
        }), () => {
            setTimeout(() => {
                this.setState({ ready: true });
            }, 1);
        });
    }

    render() {
        const columns = [
            {
                Header: 'Fecha',
                accessor: 'date',
            },
            {
                Header: 'Patente',
                accessor: 'plate',
            },
            {
                Header: 'Código',
                accessor: 'code',
            },
            {
                Header: 'Cliente',
                accessor: 'name',
            },
            {
                Header: 'Estado',
                accessor: 'status',
            },
            {
                Header: 'Ver',
                id: 'actions',
                accessor: (row) => null,
                filterable: false,
                sortable: false,
                maxWidth: 50,
                Cell: (row) => (
                    <div className="form-actions">
                        <span className="form-actions__icon" onClick={this.toggle}><i className="fas fa-eye" /></span>
                    </div>
                ),
            },
        ];

        const { showing, ready } = this.state;

        const styles = StyleSheet.create({
            page: {
                flexDirection: 'row',
                backgroundColor: '#FFF',
            },
            section: {
                margin: 10,
                padding: 10,
                flexGrow: 1,
                fontSize: 100,
                fontWeight: 700,
                color: '#CCC',
            },
            text: {
                color: '#F00',
                margin: 10,
                padding: 10,
                flexGrow: 1,
                fontSize: 12,
            },
        });

        const doc = (
            <Document>
                <Page size="A4" style={styles.page}>
                    <View style={styles.section}>
                        <Text>PDF Icafal</Text>
                    </View>
                </Page>
            </Document>
        );

        return (
            <>
                <TopBar>
                    <IconButton onClick={() => { }} icon="fas fa-file-download" />
                </TopBar>

                <ReactTable
                    data={dummyData}
                    columns={columns}
                    {...tableConfig}
                />

                {/* {docviewr} */}

                <Modal isOpen={showing} toggle={this.toggle}>
                    <ModalHeader toggle={this.toggle}>
                        Máquina 1...
                    </ModalHeader>
                    <ModalBody>
                        {ready && (
                            <PDFViewer width="100%" height="300px">
                                {doc}
                            </PDFViewer>
                        )}
                    </ModalBody>
                    <ModalFooter>
                        {ready && (
                            <PDFDownloadLink
                                document={doc}
                                fileName="test.pdf"
                                style={{
                                    textDecoration: 'none',
                                    padding: '10px',
                                    color: '#FFF',
                                    backgroundColor: '#FF5900',
                                    borderRadius: '3px',
                                }}
                            >
                                {
                                    ({ blob, url, loading, error }) =>
                                        (loading ? 'Cargando documento...' : 'Descargar')
                                }
                            </PDFDownloadLink>
                        )}
                    </ModalFooter>
                </Modal>
            </>
        );
    }
}

export default History;
