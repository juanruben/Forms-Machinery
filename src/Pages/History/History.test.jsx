import React from 'react';
import { shallow } from 'enzyme';
import History from './History';

test('should render ok', () => {
    const wrapper = shallow(<History />);
    expect(wrapper.debug()).toMatchSnapshot();
});
