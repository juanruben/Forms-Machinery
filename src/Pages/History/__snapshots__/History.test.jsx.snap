// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ok 1`] = `
"<Fragment>
  <TopBar>
    <IconButton onClick={[Function: onClick]} icon=\\"fas fa-file-download\\" />
  </TopBar>
  <ReactTable data={{...}} columns={{...}} filterable={true} className=\\"-striped\\" defaultPageSize={10} previousText=\\"<\\" nextText=\\">\\" loadingText=\\"Cargando\\" noDataText=\\"Sin datos\\" pageText=\\"\\" ofText=\\"de\\" rowsText=\\"registros\\" pageSizeOptions={{...}} resolveData={[Function: resolveData]} loading={false} showPagination={true} showPaginationTop={false} showPaginationBottom={true} showPageSizeOptions={true} defaultPage={0} showPageJump={true} collapseOnSortingChange={true} collapseOnPageChange={true} collapseOnDataChange={true} freezeWhenExpanded={false} sortable={true} multiSort={true} resizable={true} defaultSortDesc={false} defaultSorted={{...}} defaultFiltered={{...}} defaultResized={{...}} defaultExpanded={{...}} defaultFilterMethod={[Function: defaultFilterMethod]} defaultSortMethod={[Function: defaultSortMethod]} onPageChange={[undefined]} onPageSizeChange={[undefined]} onSortedChange={[undefined]} onFilteredChange={[undefined]} onResizedChange={[undefined]} onExpandedChange={[undefined]} pivotBy={[undefined]} pivotValKey=\\"_pivotVal\\" pivotIDKey=\\"_pivotID\\" subRowsKey=\\"_subRows\\" aggregatedKey=\\"_aggregated\\" nestingLevelKey=\\"_nestingLevel\\" originalKey=\\"_original\\" indexKey=\\"_index\\" groupedByPivotKey=\\"_groupedByPivot\\" onFetchData={[Function: onFetchData]} style={{...}} getProps={[Function: emptyObj]} getTableProps={[Function: emptyObj]} getTheadGroupProps={[Function: emptyObj]} getTheadGroupTrProps={[Function: emptyObj]} getTheadGroupThProps={[Function: emptyObj]} getTheadProps={[Function: emptyObj]} getTheadTrProps={[Function: emptyObj]} getTheadThProps={[Function: emptyObj]} getTheadFilterProps={[Function: emptyObj]} getTheadFilterTrProps={[Function: emptyObj]} getTheadFilterThProps={[Function: emptyObj]} getTbodyProps={[Function: emptyObj]} getTrGroupProps={[Function: emptyObj]} getTrProps={[Function: emptyObj]} getTdProps={[Function: emptyObj]} getTfootProps={[Function: emptyObj]} getTfootTrProps={[Function: emptyObj]} getTfootTdProps={[Function: emptyObj]} getPaginationProps={[Function: emptyObj]} getLoadingProps={[Function: emptyObj]} getNoDataProps={[Function: emptyObj]} getResizerProps={[Function: emptyObj]} column={{...}} expanderDefaults={{...}} pivotDefaults={{...}} pageJumpText=\\"jump to page\\" rowsSelectorText=\\"rows per page\\" TableComponent={[Function: TableComponent]} TheadComponent={[Function: cmp]} TbodyComponent={[Function: cmp]} TrGroupComponent={[Function: TrGroupComponent]} TrComponent={[Function: TrComponent]} ThComponent={[Function: ThComponent]} TdComponent={[Function: TdComponent]} TfootComponent={[Function: cmp]} FilterComponent={[Function: FilterComponent]} ExpanderComponent={[Function: ExpanderComponent]} PivotValueComponent={[Function: PivotValueComponent]} AggregatedComponent={[Function: AggregatedComponent]} PivotComponent={[undefined]} PaginationComponent={[Function: ReactTablePagination]} PreviousComponent={[undefined]} NextComponent={[undefined]} LoadingComponent={[Function: LoadingComponent]} NoDataComponent={[Function: cmp]} ResizerComponent={[Function: cmp]} PadRowComponent={[Function: PadRowComponent]} />
</Fragment>"
`;
